src/components/PostPicker/index.js

```diff
@@ -49,8 +63,8 @@ function PostList( props ) {
 						queriedPosts.map( post => (
 							<CheckboxControl
 								key={ post.id }
-								checked={ values?.includes( post.id ) }
-								label={ post.title?.rendered }
+								checked={ values.includes( post.id ) }
+								label={ post.title?.rendered || __( '(No title)', 'block-editor-components' ) }
```

Comment:
Things are nicer with spaces

```suggestion
								label={ post.title?.rendered || __( '( No title )', 'block-editor-components' ) }
```